qiling/extensions/tracing/formats/base.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+# 
+# Cross Platform and Multi Architecture Advanced Binary Emulation Framework
+# This code structure is copied and modified from the coverage extension
+
+from abc import ABC, abstractmethod
+
+
+class QlBaseTrace(ABC):
+    """
+    An abstract base class for trace collectors.
+    To add support for a new coverage format, just derive from this class and implement
+    all the methods marked with the @abstractmethod decorator.
+    """
+    
+    def __init__(self):
+        super().__init__()
+
+    @property
```

Comment:
Aggregating `property` on top of `staticmethod` doesn't work as a static property anymore.
See: https://stackoverflow.com/questions/58871322/property-with-staticmethod-in-python-3 (see references on top)